### PR TITLE
C-4972: Expose bell rendering via props

### PR DESCRIPTION
**Pull request by tk26**
## Description

Expose bell via props

Storybook screenshot where bell was custom rendered:
Screen Shot 2022-01-26 at 5 46 36 AM

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 1]
```diff
@@ -245,6 +245,10 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
 
   const isMobile = windowSize?.width ? windowSize?.width <= 480 : false;
 
+  const BellWrapper = () => {
+    return props.renderBell ? props.renderBell({}) ?? bell : bell;
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Inbox/index.tsx:249]: we need to pass all the same props that get assigned to the current bell to this renderBell function

File: packages/react-inbox/README.md [hunk 2]
```diff
@@ -81,6 +81,7 @@ interface InboxProps = {
   placement?: "top" | "left" | "right" | "bottom";
 
   // Render Props for Custom Rendering
+  renderBell?: React.FunctionComponent;
```
Comment (rileylnapier) [inline packages/react-inbox/README.md:84]: can we add the props to the type definition?

File: packages/storybook/stories/inbox/custom-components.tsx [hunk 3]
```diff
@@ -23,3 +23,7 @@ export const CustomTabs: React.FunctionComponent = () => {
 export const CustomNoMessages: React.FunctionComponent = () => {
   return <div>No Messages</div>;
 };
+
+export const CustomBell: React.FunctionComponent = () => {
+  return <div>Bell</div>;
```
Comment (rileylnapier) [inline packages/storybook/stories/inbox/custom-components.tsx:28]: can this verify the props?

Comment (tk26) [reply packages/storybook/stories/inbox/custom-components.tsx:28]: Not sure how we can do it here? I console logged in local and could see it -
Screen Shot 2022-01-28 at 10 57 25 AM

**Review by rileylnapier — APPROVED** [review]

**Pull request merged**